charts/opentelemetry-collector/templates/_config.tpl

```diff
@@ -53,8 +63,14 @@ logs:
 {{- $_ := set $config.service.pipelines.metrics "receivers" (mustWithout $config.service.pipelines.metrics.receivers "prometheus") }}
 {{- $_ := unset $config.service.telemetry.metrics "address" }}
 {{- $_ := set $config.service "telemetry" (mustMerge $config.service.telemetry (include "opentelemetry-collector.otelsdkotlp.metrics" . | fromYaml)) }}
+{{- else if .Values.config.service.telemetry.metrics.address}}
+{{/* First replace env: with env! so we can split the host with the port and replace it back later */}}
+{{- $address:= .Values.config.service.telemetry.metrics.address | replace "{env:" "{env!" | split ":" }}
+{{- $_ := unset $config.service.telemetry.metrics "address" }}
+{{- $_ := set $config.service "telemetry" (mustMerge $config.service.telemetry (include "opentelemetry-collector.metrics.prometheus" $address | fromYaml)) }}
 {{- end }}
 {{- if .Values.internalTelemetryViaOTLP.logs.enabled }}
+{{- $_ := unset $config.service.telemetry.metrics "address" }}
```

Comment:
I don't believe we need to mess with the metrics address in this logs scenario.

Reply:
sry slipped in :(